go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go

```diff
@@ -503,6 +505,20 @@ func (nadController *NetAttachDefinitionController) addNADToController(ncm Netwo
 						nadController.name, nadName, networkName)
 				}
 			}
+			if nni.nadInDeletion != "" {
+				if nni.nadInDeletion == nadName {
+					// the NAD being added is exactly the same as NAD being deleted, keep trying to delete the old NAD first
+					err = nadController.deleteNADFromControllerCommon(nni, networkName, nadName)
+					if err != nil {
+						return err
```

Comment:
does this controller have retries and so if this fails again you will keep retrying?

Reply:
yes, it will keep trying to add the new NAD if this fails.